src/realm/index_string.cpp

```diff
@@ -1300,6 +1482,78 @@ bool StringIndex::is_empty() const
     return m_array->size() == 1; // first entry in refs points to offsets
 }
 
+template <>
+void StringIndex::insert<StringData>(ObjKey key, StringData value)
+{
+    StringConversionBuffer buffer;
+
+    if (this->m_target_column.is_fulltext()) {
+        std::set<std::string> words = tokenize(value);
+
+        for (auto& word : words) {
```

Comment:
A `vector` is likely going to be faster here.

Reply:
Answered above